### ISSUE TITLE
Chrome 49 beta does not play Flash HLS for OSMF-based players anymore

### ISSUE THREAD

**Issue by arnaudleyder** [mangui/flashls#489]
I just tried Chrome beta 49.0.2623.47 with Flash HLS on an OSMF-based player and I am unable to play the stream due to a MediaErrorEvent.MEDIA_ERROR event (Message: Stream not found). This happens not at Flash load but when I first click play. Same thing with Chrome Canary 50.

This can be reproduced on http://www.flashls.org/latest/examples/osmf/GrindPlayer.html or http://www.flashls.org/latest/examples/osmf/StrobeMediaPlayback.html. I did reproduce the issue on Windows 10 and Windows 8.1. The issue seems to come from Flash 21 which Chrome 49 beta is currently sporting (I did reproduce the issue with Firefox Flash 21 beta). 

This issue does not seem to appear with video.js or chromeless players. I am not seeing this issue with Strobe when streaming MP4 in progressive download on Chrome 49 beta.
Are you able to reproduce? Do you have any idea on what could be the cause of this issue?

Thanks
Arnaud

**Comment by mangui** [comment]
indeed I can also reproduce on Chrome 50 / OS X / Flash 21. 
Chromeless player is working fine but both SMP/GrindPlayer are failing.
I investigated and it is failing here:
https://github.com/mangui/flashls/blob/dev/src/org/mangui/osmf/plugins/loader/HLSLoaderBase.as#L46

`if (urlResource.url.search(/(https?|file)\:\/\/.*?\m3u(\?.*)?/i) !== -1) {`
returns -1 on http://www.streambox.fr/playlists/x36xhzz/x36xhzz.m3u8 with Flash 21 ???? WTF

if you replace this line by
` if (urlResource.url.indexOf(".m3u") > 0) {

it works as expected.

**Comment by mangui** [comment]
maybe we can replace the regexp by 

```
if (
  (urlResource.url.indexOf("http") === 0 || urlResource.url.indexOf("file") === 0) &&
  (urlResource.url.indexOf(".m3u") > 0)
) 
```

but this is still puzzling and not satisfactory

**Comment by arnaudleyder** [comment]
Or 

```
var testM3u:RegExp = /(https?|file)\:\/\/.*?\m3u(\?.*)?/i;
if (testM3u.test(urlResource.url))
```

To be closer to what you have. Still if the search method is broken in Flash 21 that is for sure a problem.

**Comment by mangui** [comment]
if `Regexp.text` works sure

**Comment by mangui** [comment]
no success with

```
var testM3u:RegExp = /(https?|file)\:\/\/.*?\m3u(\?.*)?/i;
if (testM3u.test(urlResource.url))
```

**Comment by mangui** [comment]
looks like there are significant regressions on Regexp :-( 
Regexp are also used for trimming/simple replacement in Manifest.as

**Comment by arnaudleyder** [comment]
Yes I am seeing the same thing. Some patterns seem to pass the test (`/(https?|file)\:\/\//i` works) but some do not (`/\m3u8/i`does not work). I am not sure what is going wrong.

**Comment by arnaudleyder** [comment]
Just tried Flash beta 21.0.0.157 on FF this morning and whatever was wrong with previous Flash 21 beta releases regarding RegExp  seems to be fixed now. Chrome 49 should update soon I hope.
